uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/sdkmqtt/MqttConnectionImpl.java

```diff
@@ -198,12 +206,22 @@ public void disconnect(long timeout, int reasonCode) throws MqttException {
             client.stop(disconnectPacket);
 
             try {
+                final long deadline = System.nanoTime() + timeout * 1_000_000_000;
                 lifecycleEvents.stoppedFuture.get(timeout, TimeUnit.SECONDS);
+
+                long remaining = deadline - System.nanoTime();
+                if (remaining < MIN_SHUTDOWN_NS) {
+                    remaining = MIN_SHUTDOWN_NS;
+                }
+
+                executorService.shutdown();
+                executorService.awaitTermination(remaining, TimeUnit.NANOSECONDS);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Check the return value of `awaitTermination()` to determine if the operation timed out while waiting for other threads to stop executing. Handle when `awaitTermination()` times out. See the following diffs on GitHub that make this code change: [diff1](https://github.com/voldemort/voldemort/commit/e44ec88c57c1293a5b71598e0ad0fed403387744#diff-c220173cebaeedd4ef7bad06e679a0938710546ed77afac1d37a3ddf25e30e97L533), [diff2](https://github.com/utPLSQL/utPLSQL-cli/commit/079e021a747fe3829f868ad08c95a6eef2cc2f1c#diff-8a4bc1fbdd325f803130235d0cc70d6ae0d4b491b7440263c4f8a55452fd45eeL204)

Learn more: [ExecutorService.awaitTermination in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html#awaitTermination-long-java.util.concurrent.TimeUnit-)

Reply:
Will be fixed in next PR